### PR TITLE
Update CI pipeline.

### PR DESCRIPTION
**Pull request by steveworley**
- Moves to the community `docker:cli` image
- Moves to the community `docker:dind` image

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -54,6 +54,12 @@ workflow:
     - if [ -z "$DEPLOY_TAG" ]; then echo "DEPLOY_TAG must be set for an actionable build."; exit 1; fi
     # Latest tags need a confirmation var (provided manually: DEPLOY_LATEST)
     - if [[ "$DEPLOY_TAG" == "10.x-latest" ]] && [ -z "$DEPLOY_LATEST" ]; then echo "DEPLOY_LATEST must be set to progress with 'latest' tags."; exit 1; fi
+    # Install ahoy into the runner image.
+    - |
+      os=$(uname -s | tr [:upper:] [:lower:]) && architecture=$(case $(uname -m) in x86_64 | amd64) echo "amd64" ;; aarch64 | arm64 | armv8) echo "arm64" ;; *) echo "amd64" ;; esac) \
+        && wget -q https://github.com/ahoy-cli/ahoy/releases/latest/download/ahoy-bin-$os-$architecture -O /usr/local/bin/ahoy \
+        && chmod +x /usr/local/bin/ahoy
+    - apk add bash
```
Comment (steveworley) [inline .gitlab-ci.yml]: Bash is required for `ahoy` with default configuration; we can change this with `entrypoint` configuration however to keep this as similar as possible we'll install bash for now.

File: .ahoy.yml [hunk 2]
```diff
@@ -51,51 +51,51 @@ commands:
...
     usage: Login to a website.
-    cmd: docker-compose exec -T cli drush uli "$@"
+    cmd: docker compose exec -T cli drush uli "$@"
 
   pull:
     usage: Pull latest docker images.
     cmd: docker image ls --format \"{{.Repository}}:{{.Tag}}\" | grep govcms/ | grep -v none | xargs -n1 docker pull | cat
 
   audit-site:
     usage: Run site audit
```
Comment (yusufhm) [inline .ahoy.yml]: I believe drutiny does not exist anymore in the image. So this could be removed?
```suggestion
```

File: .ahoy.yml [hunk 3]
```diff
@@ -51,51 +51,51 @@ commands:
...
-    cmd: docker-compose exec -T cli drutiny profile:run d8-full @self "$@"
+    cmd: docker compose exec -T cli drutiny profile:run d8-full @self "$@"
```
Comment (yusufhm) [inline .ahoy.yml]:
```suggestion
```

**Review by yusufhm — COMMENTED** [review]
PR looks good - there's just one of the commands, `audit-site` which I think can be removed altogether since `drutiny` is not in the image.

**Pull request merged**